plan.py

```diff
@@ -150,7 +150,8 @@ def find_boms(self, inputs=None):
             if input_.product.boms:
                 product_bom = input_.product.boms[0].bom
                 res.append((input_.product.id, product_bom.id))
-                res += self.find_boms(product_bom.inputs)
+                if product_bom.id not in (x[1] for x in res if x):
```

Comment:
In order to really avoid recursion errors we must keep a list of the products processed in all iteration levels.

Reply:
res variable contains all boms in all iteration levels (by level execution) , this avoid recursion because do not call find_boms for already processed boms.

Reply:
Of course, you're right!

BTW do we need the "if x" in the expression "x[1] for x in res if x".